src/main/java/co/cask/PluginJSON.java

```diff
@@ -196,6 +216,63 @@ public void execute()  throws MojoExecutionException {
   }
 
   /**
+   * Inspects all the Icon files to generate properties to be a part of the related widget.
+   */
+  private JSONObject getIcon(List<String> fileNames, File[] files, File nameFile)
+    throws MojoExecutionException {
+
+    JSONObject object = new JSONObject();
+    String name = FileUtils.removeExtension(nameFile.getName());
+    if (fileNames.contains(name)) {
+      object.put("type", "inline");
+      Map<String, String> arguments = new TreeMap<String, String>();
+      File file = files[fileNames.indexOf(name)];
+      try {
+        arguments.put("data", getDataURISchemaAsString(file, "data", "image/" + FileUtils.getExtension(
+          file.getName()), true));
+      } catch (Exception e) {
+        throw new MojoExecutionException(
```

Comment:
in general, try to avoid wrapping exceptions if possible. In this case, just throw MojoExecutionException from the method above